### ISSUE TITLE
1.20 "mod failed to load correctly"

### ISSUE THREAD

**Issue by CookieMan3211** [Minecraft-Transit-Railway/Minecraft-Transit-Railway#701]
Yo, im trying to install this mod in 1.20 to play with my friends. The server crashes when im starting it I got  Architectury api. Im not getting any crash reports it just says "mod failed to load correctly" can anyone help me? My friends have been waiting for almost half a month, please.

**Comment by kettle-7** [comment]
For now I would suggest to just use 1.19.x as the 1.20 version of MTR is experimental, early in development and it takes a lot of time to port a mod this big to a new Minecraft version, especially one with quite a few under-the-hood changes.

**Comment by sandalle** [comment]
Here's the crash:
```
[15:50:35] [main/ERROR] [ne.mi.fm.lo.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/gui/Gui for invalid dist DEDICATED_SERVER
```

Error log:
```
        Crash Report UUID: 30a88bdb-d52e-47d4-84a8-9beb18a7525b
        FML: 47.0
        Forge: net.minecraftforge:47.0.35[15:50:51] [main/ERROR] [minecraft/Main]: Failed to start the minecraft server
net.minecraftforge.fml.LoadingFailedException: Loading errors encountered: [
        Minecraft Transit Railway (mtr) has failed to load correctly
§7java.lang.BootstrapMethodError: java.lang.RuntimeException: Attempted to load class net/minecraft/client/gui/GuiGraphics for invalid dist DEDICATED_SERVER
]
        at net.minecraftforge.fml.ModLoader.waitForTransition(ModLoader.java:246) ~[fmlcore-1.20.1-47.0.35.jar%23544!/:?] {}
        at net.minecraftforge.fml.ModLoader.lambda$dispatchAndHandleError$20(ModLoader.java:210) ~[fmlcore-1.20.1-47.0.35.jar%23544!/:?] {}
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?] {re:mixin}
        at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:210) ~[fmlcore-1.20.1-47.0.35.jar%23544!/:?] {}
        at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$13(ModLoader.java:183) ~[fmlcore-1.20.1-47.0.35.jar%23544!/:?] {}
        at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {re:mixin}
        at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:183) ~[fmlcore-1.20.1-47.0.35.jar%23544!/:?] {}
        at net.minecraftforge.server.loading.ServerModLoader.load(ServerModLoader.java:32) ~[forge-1.20.1-47.0.35-universal.jar%23548!/:?] {re:classloading}
        at net.minecraft.server.Main.main(Main.java:125) ~[server-1.20.1-20230612.114412-srg.jar%23543!/:?] {re:classloading}
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?] {}
        at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111) ~[fmlloader-1.20.1-47.0.35.jar%2369!/:?] {}
        at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.serverService(CommonLaunchHandler.java:103) ~[fmlloader-1.20.1-47.0.35.jar%2369!/:?] {}
        at net.minecraftforge.fml.loading.targets.CommonServerLaunchHandler.lambda$makeService$0(CommonServerLaunchHandler.java:27) ~[fmlloader-1.20.1-47.0.35.jar%2369!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar%2355!/:?] {}
        at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
```

Full crash-report: [crash-2023-07-03_15.50.51-fml.txt](https://github.com/jonafanho/Minecraft-Transit-Railway/files/11941751/crash-2023-07-03_15.50.51-fml.txt)

**Comment by jonafanho** [comment]
This should be fixed in 4.0.0. Please reopen this bug if you still encounter this issue with 4.0.0, thanks!
